manifests/machineconfigcontroller/machineconfigurationpool-worker-validatingadmissionpolicy.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "machine-configuration-pool-selector"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    matchPolicy: Equivalent
+    namespaceSelector: {}
+    objectSelector: {}
+    resourceRules:
+    - apiGroups:   ["machineconfiguration.openshift.io"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE","UPDATE"]
+      resources:   ["machineconfigpools"]
+      scope: "*"
+  validations:
+    - expression: (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "master") ||
+        (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "worker") ||
+        (object.spec.machineConfigSelector.matchExpressions.exists(e, e.key == "machineconfiguration.openshift.io/role" && e.operator == "In" && "worker" in e.values))
+      message: "MachineConfigPool must have a 'machineconfiguration.openshift.io/role' label set to include 'worker' if it is not a 'master'."
```

Comment:
I think the logic here makes sense, but it might be useful to check if the fields(`matchLabels` or `matchExpressions`) exist before using them in the condition. If any errors show up while evaluating the expression, the policy will error out and not propagate the error correctly. It'll still cause a denial, but won't be helpful to the user. For example, testing this on my cluster:
```
$ oc apply -f infra-pool.yaml 
The machineconfigpools "worker-t" is invalid: : ValidatingAdmissionPolicy 'machine-configuration-pool-selector' with binding 'machine-configuration-pool-selector-binding' denied request: expression '(object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "master") || (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "worker") || (object.spec.machineConfigSelector.matchExpressions.exists(e, e.key == "machineconfiguration.openshift.io/role" && e.operator == "In" && "worker" in e.values))' resulted in error: no such key: matchExpressions

```
The pool I tested with was the one defined the bug.

Reply:
```suggestion
      message: "All custom MachineConfigPools must inherit from the worker pool and must have a machineConfigSelector that matches for a 'machineconfiguration.openshift.io/role: worker' label " 
```

Reply:
agreed